pkgs/development/haskell-modules/generic-builder.nix

```diff
@@ -106,22 +106,22 @@ let
   crossCabalFlagsString =
     stdenv.lib.optionalString isCross (" " + stdenv.lib.concatStringsSep " " crossCabalFlags);
 
-  defaultConfigureFlags = [
+  defaultConfigureFlags = flatten [
```

Comment:
I don't like this change. Having a list of optional strings works just fine, IMHO, and I don't see what we gain by turning that into a list of lists that will be flattened. That feels like an unnecessary complication, and the change is certainly off-topic for this PR, which says it's about improving GHCJS support.